protocol/0014-ORDT-order_types.md

```diff
@@ -90,6 +90,76 @@ Network orders are used during [position resolution](./0012-POSR-position_resolu
+
+### Iceberg Order Execution
+
+1. An aggressive iceberg order that crosses with an order where volume > iceberg volume, the iceberg order gets fully filled on entry, the iceberg order status is filled, the remaining quantity = 0. Are atomic trades generated OR one single trade for each display quantity volume ???
+2. An aggressive iceberg order that crosses with an order where volume < iceberg volume. The initial display quantity is filled and the remaining volume is unfilled. Status of iceberg order is partially filled , the volume remaining = (quantity - initial volume) and the remaining volume sits on the book. When additional orders thar are submitted which consume the remaining volume on the iceberg order , the volume of the iceberg order is refreshed as and when the volume dips below the minimum peak size.
+3. A passive iceberg order (the only order at a particular price level) when crossed with another order that comes in which consumes the full volume of the iceberg order is fully filled. Status of iceberg order is filled and the remaining = 0. Atomic trades are produced.
+4. A passive iceberg order with a couple of order that sit behind the iceberg order at the same price that crosses with an order where volume > display quantity of iceberg order. After the first trade is produced , the iceberg order is pushed to the back of the queue and gets filled only when the other orders in front get fully filled.
+5. Submit an aggressive iceberg order for say size 100. There are multiple matching orders of size 30,40,50. Ensure the orders are matched and filled in time priority of the orders and any remaining volume on the orders is correctly left behind.
+6. Submit an aggressive iceberg order for say size 100. There are multiple matching orders of size 20,30. Ensure the orders are matched and filled in time priority of the orders. Ensure remaining volume on the iceberg order is (100 - (20+30))
+7. When a non iceberg order sitting on the book is amended such that it trades with with an iceberg order, then the iceberg order is refreshed.
+8. Try for scenarios for wash trading of iceberg orders - is that even possible ? Or some complex scenarios - same party , one iceberg order that sits at the back of the queue , another normal order in opposite direction , the iceberg at the back slowly comes in front and matches either fully OR partially - is this possible ?
+9. What other cases need adding ?
+
+### Snapshots
+
+1. All data pertaining to iceberg orders is saved and can be restored using the snapshot.
+
+### API
+
+1. API end points should be available to query initial peak size, minimum peak size, quantity, displayed quantity and remaining.
+2. The additional fields relating to iceberg orders should be available in the streaming api end points
+
+### Protocol Upgrade
+
+1. ???
+
+### Regression
+
+1. All other order types should behave as they were
```

Comment:
Not necessary, other Acs continuing to pass should be the criteria for this already.

Reply:
Removed this section for the AC's.